src/components/EpOsaamismerkki/EpOsaamismerkkiModal.vue

```diff
@@ -0,0 +1,324 @@
+<template>
+  <b-modal class="backdrop"
+           id="osaamismerkkiModal"
+           ref="osaamismerkkiModal"
+           :no-close-on-backdrop="true"
+           :no-enforce-focus="true"
+           :lazy="true"
+           size="xl"
+           :hide-footer="true">
+    <template slot="modal-header">
+      <div class="row w-100">
+        <div class="col">
+          <span v-if="osaamismerkki.id" class="mr-2">{{ $t('muokkaa-osaamismerkkiä')}}</span>
+          <span v-else class="mr-2">{{ $t('lisaa-osaamismerkki')}}</span>
+        </div>
+        <div>
+          <EpKielivalinta/>
+        </div>
+        <div class="close-btn clickable ml-3 pt-1" @click="sulje">
+          <EpMaterialIcon aria-hidden="false" :aria-label="$t('sulje')">close</EpMaterialIcon>
+        </div>
+      </div>
+    </template>
+
+    <div class="mb-2">
+      <b-form-group :label="$t('tila')">
+        <div class="d-flex">
+          <b-form-checkbox v-model="isJulkinen">
+            {{ $t('naytetaan-julkisena') }}
+          </b-form-checkbox>
+          <span v-if="osaamismerkki.muokattu" class="muokattu-text ml-1">{{ muokkausText + $sdt(osaamismerkki.muokattu)}}</span>
+        </div>
+      </b-form-group>
+
+      <b-form-group :label="$t('nimi') + ' *'">
+        <EpField v-model="osaamismerkki.nimi" :is-editing="true" :validation="$v.osaamismerkki.nimi" :showValidValidation="false"/>
+      </b-form-group>
+
+      <b-form-group :label="$t('kategoria') + ' *'">
+        <EpMultiSelect :is-editing="true"
+                       :options="osaamismerkkiKategoriat"
+                       v-model="osaamismerkki.kategoria"
+                       :placeholder="$t('kaikki')">
+          <template v-slot:singleLabel="{ option }">{{ $kaanna(option.nimi) }}</template>
+          <template v-slot:option="{ option }">{{ $kaanna(option.nimi) }}</template>
+          <template v-slot:tag="{ option }">{{ $kaanna(option.nimi) }}</template>
+        </EpMultiSelect>
+      </b-form-group>
+
+      <b-form-group :label="$t('voimassaolo') + ' *'">
+        <div class="d-flex align-items-center">
+          <EpDatepicker v-model="osaamismerkki.voimassaoloAlkaa" :is-editing="true"/>
+          <div class="ml-2 mr-2">-</div>
+          <EpDatepicker v-model="osaamismerkki.voimassaoloLoppuu" :is-editing="true"/>
+        </div>
+      </b-form-group>
+    </div>
+
+    <b-form-group :label="$t('osaamistavoitteet') + ' *'">
+      <draggable v-bind="defaultDragOptions"
+                 tag="div"
+                 v-model="osaamistavoitteet">
+        <div class="row mb-2" v-for="(tavoite, i) in osaamistavoitteet" :key="'tavoite'+i">
```

Comment:
Eikös tähän voi suorilta laittaa "in osaamismerkki.osaamistavoitteet" jotta ei tarvitse erikseen getteriä ja setteriä ?